### PR TITLE
Docs: update "AiiDA internals for developers" to `aiida-core==1.0.0`

### LINKED ISSUE #3173: Check documentation for "for AiiDA developers/AiiDA internals"
Check if the section is updated to AiiDA 1.0.

https://aiida.readthedocs.io/projects/aiida-core/en/latest/developer_guide/core/internals.html

### PR DESCRIPTION
**Pull request by sphuber**
Fixes #3173

### CODE REVIEW

**Comment by giovannipizzi** [comment]
Yes, let's just check if there are major problems.
Then if you feel the text can be improved, I suggest that you prepare a second PR on top of this (and for the review of these textual PRs, it might be much faster to sit down in person with the reviewer to do the modifications on the fly, so the PR is already ready to be merged).

File: docs/source/developer_guide/core/internals.rst [hunk 1]
```diff
@@ -5,211 +5,221 @@ AiiDA internals
 Node
 ++++
 
-The :py:class:`~aiida.orm.nodes.Node` class is the basic class that represents all the possible objects at the AiiDA world. More precisely it is inherited by many classes including (among others) the :py:class:`~aiida.orm.nodes.process.ProcessNode` class, representing computations that convert data into a different form, the :py:class:`~aiida.orm.nodes.data.code.Code` class representing executables and file collections that are used by calculations and the :py:class:`~aiida.orm.nodes.data.data.Data` class which represents data that can be input or output of calculations.
+The :py:class:`~aiida.orm.nodes.node.Node` class is the basic class that represents all the possible objects at the AiiDA world.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/developer_guide/core/internals.rst]: Done

File: docs/source/developer_guide/core/internals.rst [hunk 2]
```diff
@@ -5,211 +5,221 @@ AiiDA internals
...
+More precisely it is inherited by many classes including (among others) the :py:class:`~aiida.orm.nodes.process.ProcessNode` class, representing computations that convert data into a different form, the :py:class:`~aiida.orm.nodes.data.code.Code` class representing executables and file collections that are used by calculations and the :py:class:`~aiida.orm.nodes.data.data.Data` class which represents data that can be input or output of calculations.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/developer_guide/core/internals.rst]: Done

File: docs/source/developer_guide/core/internals.rst [hunk 3]
```diff
@@ -5,211 +5,221 @@ AiiDA internals
...
 
 
 Immutability concept
 ********************
-A node can store information through attributes. Since AiiDA guarantees a certain level of provenance, these attributes become immutable as soon as the node is stored.
-This means that as soon as a node is stored any attempt to alter its attributes, changing its value or deleting it altogether, shall be met with a raised exception.
-Certain subclasses of nodes need to adapt this behavior however, as for example in the case of the :py:class:`~aiida.orm.nodes.process.ProcessNode` class (see `calculation updatable attributes`_), but since the immutability
-of stored nodes is a core concept of AiiDA, this behavior is nonetheless enforced on the node level. This guarantees that any subclasses of the Node class will respect this behavior unless it is explicitly overriden.
+A node can store information through attributes.
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply docs/source/developer_guide/core/internals.rst]: Done

**Comment by CasperWA** [comment]
@ltalirz: I don't have write permission to Seb's repo, so I cannot commit your suggestions. However, I have a local branch now with both his commit and one for your changes. Should I make a new PR with it that can be merged with the correct authors - closing this one - or should we wait until Seb comes back/randomly provides me with writing permissions to his fork? :)

**Comment by ltalirz** [comment]
Sure, just make a new PR and close that one - perhaps squash it into two commits, one for seb and one for me/you, so that authorship of his changes is maintained

**Comment by CasperWA** [comment]
Redone the PR (for editing reasons) in #3271

**Comment by sphuber** [comment]
In this case, I just made sure that everything at least is correct and only rewrote things when really necessary. But still, in general, I think a lot more improvement is needed. However, this will require a reorganization of more than just this section. This is a clear case of a part of the documentation that was one of the first parts to be written and things have been added over time. An overall coherent structure is now kind of missing. Tackling this is a bigger problem then just making sure there are no inaccuracies however.

**Pull request closed without merging**